codeforamerica/brigade
# Finish story #24781887

Anyone can see a subset of the civic commons information on an app show
page
- Add nid to application model to serve as the hook to the civic commons
  api
- Add faraday to gemfile to act as middleware to civic commons
- class and spec for handling the update of civic commons data
- Spec out and write libs for wrapping civic commons API
- Create a civic commons client to retrieve establish a connection
  and retrieve/pass specific resources (nodes and taxonomy_terms)
- Create a civic commons node to handle retrieving a node's info (body in request)
- Build an application and creator handler to inherit from node as they
  are types of nodes in the civic commons api
- Add hashie, faraday_middleware to gemfile to help consume the
  civic commons API
- include specs to test out lib
- build job to handle updating all the attributes
  - Add description, creator, and license to be updated with updater
- Add the ability to retrieve taxonomy terms in the civiccommons client
- Add class and spec to handle fetching data from a taxonomy term body
- Handle exceptions thrown by the node handler by returning nil when
  body format is different than expected
- Handle case where the title returns something other than the hash
- Add cassettes
- Add application cases to provide proper test coverage for api handler
- use app_decorator to select which description will be shown on the
  show page
- make the nid the only required field
- create rake task to handle the fetching of civic commons data in the
  background by calling the previously mentioned job
- modify show and index pages to display the civic commons app data

Other
- remove home controller since we weren't doing anything with it
- Setup VCR for rspec
